src/main/java/picocli/CommandLine.java

```diff
@@ -12377,6 +12437,16 @@ private void processArguments(List<CommandLine> parsedCommands,
                 // A single option may be without option parameters, like "-v" or "--verbose" (a boolean value),
                 // or an option may have one or more option parameters.
                 // A parameter may be attached to the option.
+                if (commandSpec.parser().abbreviatedOptionsAllowed()) {
+                    Set<String> aggregatedOptionNames = new LinkedHashSet<String>();
+                    aggregatedOptionNames.addAll(commandSpec.optionsMap().keySet());
+                    aggregatedOptionNames.addAll(commandSpec.negatedOptionsMap().keySet());
+                    try {
+                        arg = AbbreviationMatcher.match(aggregatedOptionNames, arg);
+                    } catch (IllegalArgumentException ex) {
+                        throw new ParameterException(CommandLine.this, "Error: " + ex.getMessage(), ex);
```

Comment:
We can pass the `CommandLine` instance as a parameter to the `match` method so it can throw a `ParameterException` instead of an `IllegalArgException`

Reply:
That kind of breaks the encapsulation?

Reply:
You mean the `CommandLine.Interpreter::tryConvert` method, right? (line numbers may have changed). Good point!

One big difference is that `ITypeConverter` is a client-facing interface: it is intended to be implemented by users of the library. So we want to make it as simple and easy as possible: "given a String, produce a different value, or throw any exception you want". This interface would be a lot less easy to use if applications could only throw a `ParameterException`, so we put some extra effort in the framework to make it easier for applications.

The `match` method is internal, and is not meant to become public API. It will not be used outside of `CommandLine`, so I am okay with less encapsulation.